src/libraries/System.Memory/tests/Span/EnumerateRunePositions.cs

```diff
@@ -0,0 +1,248 @@
+        {
+            {
+                (Rune rune, int startIndex) = runePosition;
+                Assert.Equal(runePosition.Rune, rune);
+                Assert.Equal(runePosition.StartIndex, startIndex);
+            }
+            {
+                (Rune rune, int startIndex, int length) = runePosition;
+                Assert.Equal(runePosition.Rune, rune);
+                Assert.Equal(runePosition.StartIndex, startIndex);
+                Assert.Equal(runePosition.Length, length);
+            }
+        }
+
+        [Fact]
+        public static void RunePosition_DefaultTest()
+        {
+            RunePosition runePosition = default;
+            RunePosition_TestProps(default, 0, 0, false, runePosition);
+            RunePosition_TestEquals(default, runePosition);
+            RunePosition_TestDeconstruct(runePosition);
+
+            runePosition = new RunePosition();
+            RunePosition_TestProps(default, 0, 0, false, runePosition);
+            RunePosition_TestEquals(default, runePosition);
+            RunePosition_TestDeconstruct(runePosition);
+        }
+
+        [Theory]
+        [InlineData("x")]
+        [InlineData("\uD86D\uDF54")] // valid surrogate pair
+        [InlineData("\uFFFD")] // literal U+FFFD
+        [InlineData("\0")]
+        public static void RunePosition_UnitTest(string runeStr)
+        {
+            Rune.DecodeFromUtf16(runeStr, out Rune rune, out _);
+
+            RunePosition runePosition1 = new RunePosition(rune, 0, rune.Utf16SequenceLength, false);
+            RunePosition runePosition2 = new RunePosition(rune, 5, rune.Utf16SequenceLength, true);
+            RunePosition runePosition3 = new RunePosition(rune, 100, 5, true);
+
+            foreach (var startIndex in new [] {0, 5, 100})
+            {
+                foreach (var wasReplaced in new[] { true, false })
+                {
+                    RunePosition runePosition = new RunePosition(rune, startIndex, rune.Utf16SequenceLength, wasReplaced);
+                    RunePosition_TestProps(rune, startIndex, rune.Utf16SequenceLength, wasReplaced, runePosition);
+
+                    RunePosition_TestEquals(default, runePosition);
+                    RunePosition_TestEquals(runePosition1, runePosition);
+                    RunePosition_TestEquals(runePosition2, runePosition);
+                    RunePosition_TestEquals(runePosition3, runePosition);
+                    RunePosition_TestDeconstruct(runePosition);
+                }
+            }
+        }
+
+        [Fact]
+        public static void EnumerateRunePositions_Empty()
+        {
+            {
+                RunePosition.Utf16Enumerator enumerator = RunePosition.EnumerateUtf16(ReadOnlySpan<char>.Empty).GetEnumerator();
+                Assert.Equal(default, enumerator.Current);
+                Assert.False(RunePosition.EnumerateUtf16(ReadOnlySpan<char>.Empty).GetEnumerator().MoveNext());
+                Assert.Equal(default, enumerator.Current);
```

Comment:
Same as above: calling the `Current` property after `MoveNext` returns false is undefined behavior.